src/IO/H5/Helpers.hpp

```diff
@@ -113,9 +113,14 @@ bool contains_attribute(hid_t file_id, const std::string& group_name,
 
 /*!
  * \ingroup HDF5Group
- * \brief Read a DataVector from a dataset in a group
+ * \brief Read an array of rank 0-3 into an object (of corresponding datatype):
+ * rank 0 (double, int),
+ * rank 1 (std::vector, DataVector),
```

Comment:
You need to document the data ordering for the multiarrays. That is, how does the data on disk get mapped to the indices?